File: src/main/java/com/ject/studytrip/trip/domain/repository/TripQueryRepository.java
package com.ject.studytrip.trip.domain.repository;

import com.ject.studytrip.trip.domain.model.Trip;
import com.ject.studytrip.trip.domain.model.TripCategory;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;

public interface TripQueryRepository {
    Slice<Trip> findSliceByMemberId(Long memberId, Pageable pageable);

    long countActiveTripsByMemberIdAndCategory(Long memberId, TripCategory category);
}

File: src/main/kotlin/com/ject/studytrip/trip/domain/repository/TripQueryRepository.kt
package com.ject.studytrip.trip.domain.repository

import com.ject.studytrip.trip.domain.model.Trip
import com.ject.studytrip.trip.domain.model.TripCategory
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice

interface TripQueryRepository {
    fun findSliceByMemberIdAndCompletedFalseAndDeletedAtIsNull(
        memberId: Long,
        pageable: Pageable,
    ): Slice<Trip>

    fun countActiveTripsByMemberIdAndCategory(
        memberId: Long,
        category: TripCategory,
    ): Long
}
